santa/src/Santa.Api/Controllers/MessageController.cs

```diff
@@ -68,72 +68,85 @@ public async Task<ActionResult<Logic.Objects.Message>> PostMessage([FromBody] Ap
         {
             BaseClient logicBaseClient = await repository.GetBasicClientInformationByID(message.messageSenderClientID.GetValueOrDefault() != Guid.Empty ? message.messageSenderClientID.GetValueOrDefault() : message.messageRecieverClientID.GetValueOrDefault());
             BaseClient checkerClient = await repository.GetBasicClientInformationByEmail(User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Email).Value);
-            InfoContainer checkerInfoContainer = await repository.getClientInfoContainerByIDAsync(checkerClient.clientID);
+            InfoContainer checkerInfoContainer = new InfoContainer() { assignments = new List<Logic.Objects.RelationshipMeta>() };
+            // If the checker is an admin, no need to get the info container for the checker
+            if (!checkerClient.isAdmin)
+            {
+                checkerInfoContainer = await repository.getClientInfoContainerByIDAsync(checkerClient.clientID);
+            }
+            
 
             // If the logic client and checker client have the same Id and the relationxref is either null or part of of the checked clients assingments list (Or the checker is just an admin overall)
-            if ((logicBaseClient.clientID == checkerClient.clientID && message.clientRelationXrefID != null ? checkerInfoContainer.assignments.Any(a => a.clientRelationXrefID == message.clientRelationXrefID) : true) || checkerClient.isAdmin)
+            if (logicBaseClient.clientID == checkerClient.clientID && message.clientRelationXrefID != null)
             {
-                TimeZoneInfo easternZone = TimeZoneInfo.FindSystemTimeZoneById("Eastern Standard Time");
-
-                Logic.Objects.Message logicMessage = new Logic.Objects.Message()
+                if (checkerInfoContainer.assignments.Any(a => a.clientRelationXrefID == message.clientRelationXrefID) || checkerClient.isAdmin)

```

Comment:
I think you can bump the `checkerClient.isAdmin` check to the above `if` statement as there's no point in checking `clientIDs` and `clientRelationXrefIDs` if the user is an admin.

You can also move the check to the front of the `if` for a slighter faster path.

Reply:
That's a good suggestion, though, I have it here since the first is just checking to make sure the person grabbing the info is the person requesting the data first before moving in. Would just let me put logging in later as well between there and the next if in the future